tajo-plan/src/main/java/org/apache/tajo/plan/logical/TableSubQueryNode.java

```diff
@@ -108,14 +111,19 @@ public boolean hasTargets() {
   }
 
   @Override
-  public void setTargets(Target[] targets) {
+  public void setTargets(List<Target> targets) {
     this.targets = targets;
     setOutSchema(PlannerUtil.targetToSchema(targets));
   }
 
   @Override
-  public Target[] getTargets() {
-    return targets;
+  public List<Target> getTargets() {
+    if (hasTargets()) {
+      return targets;
```

Comment:
Yes, that's what I intended. If new errors are introduced, it means that ```targets``` are modified outside somewhere. It would be good to handle this issue in another jira if necessary.